### PR TITLE
Upgrade to JSF 2

### PR DESCRIPTION
**Pull request by solth**
Upgrade to JSF2 #742 . Contains the following steps:

- Upgrade JSF implementation from MyFaces 1.1 to MyFaces 2.2.12
- Replace JSPs with Facelets
- Replace navigation rules in `faces-config.xml` with implicit navigation
- Upgrade or replace tag libraries that are either outdated or incompatible with JSF 2 and Facelets. This step is not yet finished and will be continued in future commits.

### CODE REVIEW

File: Kitodo/src/main/webapp/newpages/inc/head.xhtml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        <t:stylesheet media="all" path="/css/toggle.js.css"/>
+        <!--Styles für panelTabbedPane unter Menü Administration- Projekte -->
+        <f:loadBundle basename="messages.messages" var="msgs"/>
+        <title><h:outputText value="#{HelperForm.applicationHeaderTitle} - #{HelperForm.version}"/></title>
+        <!-- <h:outputText value="internal version number: #{HelperForm.buildVersion}"/> -->
+        <!--<h:outputScript library="primefaces" name="jquery/jquery.js" target="head" />-->
+        <script type="text/javascript" src="#{request.contextPath}/js/toggle.js"></script>
+    </h:head>
+
+</ui:composition>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc/head.xhtml]: New line at end of file is missing.

File: Kitodo/src/main/webapp/newpages/inc/tbl_Fuss.xhtml [hunk 2]
```diff
@@ -0,0 +1,103 @@
+<!--
+        *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc/tbl_Fuss.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc/tbl_Fuss.xhtml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+                                </script>
+
+                        </td>
+                    </tr>
+                </table>
+            </h:form>
+        </td>
+    </tr>
+</ui:composition>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc/tbl_Fuss.xhtml]: New line at end of file is missing.

File: Kitodo/src/main/webapp/newpages/inc/tbl_Kopf.xhtml [hunk 4]
```diff
@@ -0,0 +1,166 @@
+<!--
+    *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc/tbl_Kopf.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc/tbl_Kopf.xhtml [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+                    </h:form>
+
+
+                </td>
+            </tr>
+        </table>
+    </td>
+    </tr>
+</ui:composition>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc/tbl_Kopf.xhtml]: New line at end of file is missing.

File: Kitodo/src/main/webapp/newpages/inc/tbl_Navigation.xhtml [hunk 6]
```diff
@@ -0,0 +1,332 @@
+<!--
+        *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc/tbl_Navigation.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/Schritte_Liste.xhtml [hunk 7]
```diff
@@ -1,54 +1,53 @@
...
-<%-- Datentabelle --%>
-<x:dataTable id="auflistung" styleClass="standardTable" width="100%" cellspacing="1px" cellpadding="1px"
-             headerClass="standardTable_Header"
-             rowClasses="standardTable_Row1,standardTable_Row2,standardTable_Row1,standardTable_Row2" var="item"
-             value="#{AktuelleSchritteForm.page.listReload}">
-
-    <%-- +++++++++++++++++  SelectionBoxes ++++++++++++++++++++++++ --%>
-    <x:column style="text-align:center" rendered="#{AktuelleSchritteForm.anzeigeAnpassen['selectionBoxes']}">
+<!--
+        *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/Schritte_Liste.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/Schritte_Liste_Action.xhtml [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+                    </h:panelGroup>
+                </h:panelGrid>
+            </td>
+        </tr>
+
+    </table>
+
+    <!-- ++++++++++++++++     // Action      ++++++++++++++++ -->
+
+</ui:composition>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/Schritte_Liste_Action.xhtml]: New line at end of file is missing.

File: Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/schritt_box_Action.xhtml [hunk 9]
```diff
@@ -0,0 +1,374 @@
...
+                            </h:panelGrid>
+                        </td>
+                    </tr>
+
+                </table>
+            </ui:fragment>
+            <!-- ++++++++++++++++     // Action      ++++++++++++++++ -->
+        </h:form>
+    </h:panelGroup>
+</ui:composition>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/schritt_box_Action.xhtml]: New line at end of file is missing.

File: Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/schritt_box_Details.xhtml [hunk 10]
```diff
@@ -0,0 +1,157 @@
+<!--
+    *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/schritt_box_Details.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/schritt_box_Properties.xhtml [hunk 11]
```diff
@@ -0,0 +1,291 @@
+<!--
+    *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/schritt_box_Properties.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/schritt_box_Properties.xhtml [hunk 12]
```diff
@@ -0,0 +1,291 @@
...
+
+                            <t:updateActionListener property="#{AktuelleSchritteForm.modusBearbeiten}" value=""/>
+                        </h:commandButton>
+
+                    </td>
+                </tr>
+            </table>
+        </ui:fragment>
+    </h:form>
+</ui:composition>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_AktuelleSchritte/schritt_box_Properties.xhtml]: New line at end of file is missing.

File: Kitodo/src/main/webapp/newpages/inc_Batches/batch_box_Properties.xhtml [hunk 13]
```diff
@@ -0,0 +1,332 @@
...
+                                         action="#{AktuelleSchritteForm.batchHelper.saveCurrentPropertyForAll}">
+                            <t:updateActionListener property="#{AktuelleSchritteForm.modusBearbeiten}" value=""/>
+                        </h:commandButton>
+                    </td>
+                </tr>
+            </table>
+        </ui:fragment>
+
+    </h:form>
+</ui:composition>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Batches/batch_box_Properties.xhtml]: New line at end of file is missing.

File: Kitodo/src/main/webapp/newpages/inc_Main/box1.xhtml [hunk 14]
```diff
@@ -0,0 +1,48 @@
+<!--
+        *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Main/box1.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_Main/box1.xhtml [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+                        <h:outputText value="#{msgs.vorlagen}:"/>
+                        <h:outputText value="#{StatistikForm.anzahlVorlagen}"/>
+                        <h:outputText value="#{msgs.werkstuecke}:"/>
+                        <h:outputText value="#{StatistikForm.anzahlWerkstuecke}"/>
+                    </h:panelGrid>
+                </td>
+            </tr>
+        </table>
+    </ui:fragment>
+</ui:composition>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Main/box1.xhtml]: New line at end of file is missing.

File: Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste.xhtml [hunk 16]
```diff
@@ -1,79 +1,78 @@
...
-    <h:outputText value="#{msgs.prozessvorlagen}" rendered="#{ProzessverwaltungForm.modusAnzeige=='vorlagen'}"/>
-    <h:outputText value="#{msgs.aktuelleProzesse}" rendered="#{ProzessverwaltungForm.modusAnzeige=='aktuell'}"/>
-</htm:h4>
-
-<%-- newUser-Schaltknopf --%>
-<h:commandLink action="#{ProzessverwaltungForm.Neu}" immediate="true"
-               rendered="#{(LoginForm.maximaleBerechtigung == 1 || LoginForm.maximaleBerechtigung == 2) && ProzessverwaltungForm.modusAnzeige!='vorlagen'}"
-               id="new1">
+<!--
+        *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Action.xhtml [hunk 17]
```diff
@@ -0,0 +1,543 @@
+<!--
+    *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Action.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Anzahlen.xhtml [hunk 18]
```diff
@@ -0,0 +1,122 @@
+<!--
+        *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Anzahlen.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Statistik.xhtml [hunk 19]
```diff
@@ -0,0 +1,245 @@
+<!--
+        *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Statistik.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Statistik.xhtml [hunk 20]
```diff
@@ -0,0 +1,245 @@
...
+                    </h:panelGroup>
+
+                    <h:outputText style="font-weight:bold" value=" - #{msgs.orLast} - "/>
+
+                    <h:inputText style="width:130px;text-align:right" required="false"
+                                 value="#{ProzessverwaltungForm.statisticsManager.sourceNumberOfTimeUnitsAsString}"/>
+
+                    <h:selectOneMenu style="width:130px" value="#{ProzessverwaltungForm.statisticsManager.sourceTimeUnit}"
+                                     converter="StatisticsTimeUnitConverter">
+                        <!--<si:selectItems value="#{ProzessverwaltungForm.statisticsManager.allTimeUnits}" var="unit"-->
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Statistik.xhtml]: Remove it if it is not needed any more.

File: Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Statistik.xhtml [hunk 21]
```diff
@@ -0,0 +1,245 @@
...
+                                                 title="#{msgs.showAverage}">
+                        </h:selectBooleanCheckbox>
+                        <h:outputLabel value="#{msgs.showAverage}"/>
+                    </h:panelGroup>
+
+                    <h:outputText value="#{msgs.einheit}" style="width:150px"/>
+
+                    <h:selectOneMenu style="width:130px" value="#{ProzessverwaltungForm.statisticsManager.targetTimeUnit}"
+                                     converter="StatisticsTimeUnitConverter">
+                        <!--<si:selectItems value="#{ProzessverwaltungForm.statisticsManager.allTimeUnits}" var="unit"-->
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Statistik.xhtml]: Remove it if is not needed any more.

File: Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Statistik.xhtml [hunk 22]
```diff
@@ -0,0 +1,245 @@
...
+                        <f:selectItems value="#{ProzessverwaltungForm.statisticsManager.allTimeUnits}" var="unit"
+                                        itemLabel="#{unit.title}" itemValue="#{unit}"/>
+                    </h:selectOneMenu>
+
+                    <h:outputText value="#{msgs.anzeige}"/>
+
+                    <h:selectOneMenu style="width:130px"
+                                     value="#{ProzessverwaltungForm.statisticsManager.targetCalculationUnit}"
+                                     converter="StatisticsCalculationUnitConverter">
+                        <!--<si:selectItems value="#{ProzessverwaltungForm.statisticsManager.allCalculationUnits}" var="unit"-->
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Statistik.xhtml]: If this is not needed any more, remove it.

File: Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Statistik.xhtml [hunk 23]
```diff
@@ -0,0 +1,245 @@
...
+                        <f:selectItems value="#{ProzessverwaltungForm.statisticsManager.allCalculationUnits}" var="unit"
+                                        itemLabel="#{unit.title}" itemValue="#{unit}"/>
+                    </h:selectOneMenu>
+
+                    <h:outputText value="#{msgs.ausgabe}"/>
+
+                    <h:selectOneMenu style="width:130px"
+                                     value="#{ProzessverwaltungForm.statisticsManager.targetResultOutput}"
+                                     converter="StatisticsResultOutputConverter">
+                        <!--<si:selectItems value="#{ProzessverwaltungForm.statisticsManager.allResultOutputs}" var="out"-->
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/Prozesse_Liste_Statistik.xhtml]: If this is not needed any more, remove it.

File: Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/prozess_box_Properties.xhtml [hunk 24]
```diff
@@ -0,0 +1,356 @@
...
+
+                    <h:commandButton value="#{msgs.uebernehmen}" action="#{ProzessverwaltungForm.saveCurrentProperty}">
+                        <t:updateActionListener property="#{ProzessverwaltungForm.modusBearbeiten}" value=""/>
+                    </h:commandButton>
+                </td>
+            </tr>
+
+        </table>
+    </ui:fragment>
+</ui:composition>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/prozess_box_Properties.xhtml]: New line at end of file is missing.

File: Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/schritt_box_Details.xhtml [hunk 25]
```diff
@@ -0,0 +1,371 @@
...
+                    document.getElementById("scripttable").style.display = "inline";
+                } else {
+                    document.getElementById("scripttable").style.display = "none";
+                }
+            }
+
+        }
+        //-->
+    </script>
+</ui:composition>
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_Prozessverwaltung/schritt_box_Details.xhtml]: New line at end of file is missing.

File: Kitodo/src/main/webapp/newpages/inc_statistic/StatisticCorrection.xhtml [hunk 26]
```diff
@@ -0,0 +1,166 @@
+<!--
+    *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_statistic/StatisticCorrection.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_statistic/StatisticProduction.xhtml [hunk 27]
```diff
@@ -0,0 +1,165 @@
+<!--
+    *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_statistic/StatisticProduction.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_statistic/StatisticStorage.xhtml [hunk 28]
```diff
@@ -0,0 +1,167 @@
+<!--
+    *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_statistic/StatisticStorage.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/inc_statistic/StatisticThroughput.xhtml [hunk 29]
```diff
@@ -0,0 +1,191 @@
+<!--
+    *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/inc_statistic/StatisticThroughput.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/modulemanager.xhtml [hunk 30]
```diff
@@ -0,0 +1,168 @@
+<!DOCTYPE html>
+<!--
+        *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/modulemanager.xhtml]: Indent of license header looks wrong.

File: Kitodo/src/main/webapp/newpages/taskmanager.xhtml [hunk 31]
```diff
@@ -0,0 +1,265 @@
+<!DOCTYPE html>
+<!--
+        *
```
Comment (henning-gerhardt) [inline Kitodo/src/main/webapp/newpages/taskmanager.xhtml]: Indent of license header looks wrong.

**Comment by henning-gerhardt** [comment]
Why did you not rename the files from `*.jsp` to `*.xhtml` in one commit without any other or only with small changes and then made other needed adjustments? This could maybe spare a lot of time for you and even me.

**Comment by solth** [comment]
Yes, you are right, I should probably just have renamed the jsp files to *.xhtml. My intention in keeping the jsp's alongside the new Facelets was to always be able to compare both versions of a template until the Facelet could completely replace the jsp.

I will make the requested adjustments and commit them as soon as possible.

File: Kitodo/src/main/webapp/newpages/MassImportPage2.xhtml [hunk 32]
```diff
@@ -0,0 +1,192 @@
...
+    >
+    <f:view locale="#{SpracheForm.locale}">
+        <ui:include src="inc/head.xhtml" />
+        <h:body>
+            <table class="headTable" cellspacing="0" cellpadding="0"
+                       style="padding-left:5px;padding-right:5px;margin-top:5px;">
+                <ui:include src="inc/tbl_Kopf.xhtml" />
+            </table>
+            <table cellspacing="5" cellpadding="0" class="layoutTable" align="center">
+                <!-- TODO: check if "h:outputStylesheet" works here! -->
```
Comment (solth) [inline Kitodo/src/main/webapp/newpages/MassImportPage2.xhtml]: This TODO is not resolved, yet, and should remain in the source code.

File: Kitodo/src/main/webapp/newpages/MassImportPage3.xhtml [hunk 33]
```diff
@@ -0,0 +1,120 @@
...
+    >
+    <f:view locale="#{SpracheForm.locale}">
+        <ui:include src="inc/head.xhtml" />
+        <h:body>
+            <table class="headTable" cellspacing="0" cellpadding="0"
+                       style="padding-left:5px;padding-right:5px;margin-top:5px;">
+                <ui:include src="inc/tbl_Kopf.xhtml" />
+            </table>
+            <table cellspacing="5" cellpadding="0" class="layoutTable" align="center">
+                <!-- TODO: check if "h:outputStylesheet" works here! -->
```
Comment (solth) [inline Kitodo/src/main/webapp/newpages/MassImportPage3.xhtml]: This TODO is not resolved, yet, and should remain in the source code.

File: Kitodo/src/main/webapp/newpages/MultiMassImportPage2.xhtml [hunk 34]
```diff
@@ -0,0 +1,196 @@
...
+    >
+    <f:view locale="#{SpracheForm.locale}">
+        <ui:include src="inc/head.xhtml" />
+        <h:body>
+            <table class="headTable" cellspacing="0" cellpadding="0"
+                       style="padding-left:5px;padding-right:5px;margin-top:5px;">
+                <ui:include src="inc/tbl_Kopf.xhtml" />
+            </table>
+            <table cellspacing="5" cellpadding="0" class="layoutTable" align="center">
+                <!-- TODO: check if "h:outputStylesheet" works here! -->
```
Comment (solth) [inline Kitodo/src/main/webapp/newpages/MultiMassImportPage2.xhtml]: This TODO is not resolved, yet, and should remain in the source code.

File: Kitodo/src/main/webapp/newpages/MassImport.xhtml [hunk 35]
```diff
@@ -0,0 +1,322 @@
...
+    >
+    <f:view locale="#{SpracheForm.locale}">
+        <ui:include src="inc/head.xhtml" />
+        <h:body>
+            <table class="headTable" cellspacing="0" cellpadding="0"
+                       style="padding-left:5px;padding-right:5px;margin-top:5px;">
+                <ui:include src="inc/tbl_Kopf.xhtml" />
+            </table>
+            <table cellspacing="5" cellpadding="0" class="layoutTable" align="center">
+                <!-- TODO: check if "h:outputStylesheet" works here! -->
```
Comment (solth) [inline Kitodo/src/main/webapp/newpages/MassImport.xhtml]: This TODO is not resolved, yet, and should remain in the source code.

**Comment by solth** [comment]
All requested changes have been implemented or postponed. 

The TODOs concerning the "h:outputStylesheet" tags are not yet solved and should remain in the source code.

The missing new lines at the end of some files have been added but for some reason Github doesn't recognize these changes and therefore the requested changed are not marked as "outdated", above.

**Pull request merged**